File: module5/thi-module5/src/app/components/employee.ts
import {EmployeeType} from './employee-type';

export class Employee {
  id: number;
  name: string;
  birthday: string;
  gender: string;
  idCard: string;
  phone: string;
  address:string;
  employeeType: EmployeeType;


  constructor(id: number, name: string, birthday: string, gender: string, idCard: string, phone: string, address: string, employeeType: EmployeeType) {
    this.id = id;
    this.name = name;
    this.birthday = birthday;
    this.gender = gender;
    this.idCard = idCard;
    this.phone = phone;
    this.address = address;
    this.employeeType = employeeType;
  }
}

File: module6/project_module6/src/main/java/com/codegym/entity/employee/Employee.java
package com.codegym.entity.employee;

import com.codegym.entity.account.Users;
import com.codegym.entity.contract.Contract;
import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonManagedReference;

import javax.persistence.*;
import java.util.Set;

@Entity
@Table(name = "employee")
public class Employee{

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;

    @Column(name = "name", nullable = false)
    private String name;

    @Column(name = "gender")
    private String gender;

    @Column(name = "date_of_birth", columnDefinition = "DATE")
    private String dateOfBirth;

    @Column(name = "id_card", nullable = false, unique = true)
    private String idCard;

    @Column(name = "phone", nullable = false, unique = true)
    private String phone;

    @Column(name = "email", nullable = false, unique = true)
    private String email;

    @Column(name = "address")
    private String address;

    @Column(name = "city")
    private String city;

    @Column(name = "img",columnDefinition = "LONGTEXT")
    private String img;
    @ManyToOne
    @JoinColumn(name = "position_id", referencedColumnName = "id")
    private Position position;

    @OneToOne
    @JsonManagedReference
    @JoinColumn(name = "user_id", referencedColumnName = "id")
    private Users users;

    @OneToMany(mappedBy = "employee", cascade = CascadeType.ALL)
    @JsonBackReference
    private Set<Contract> contractSet;

    public Employee() {
    }

    public Position getPosition() {
        return position;
    }

    public void setPosition(Position position) {
        this.position = position;
    }

    public Users getUsers() {
        return users;
    }

    public void setUsers(Users users) {
        this.users = users;
    }
}
